activerecord/lib/active_record/railtie.rb

```diff
@@ -177,14 +177,17 @@ class Railtie < Rails::Railtie # :nodoc:
             begin
               descendants.each do |model|
                 # If the schema cache was loaded from a dump, we can use it without connecting
-                schema_cache = model.connection_pool.schema_cache
-
-                # If there's no connection yet, we avoid connecting.
-                schema_cache ||= model.connected? && model.connection.schema_cache
+                if schema_reflection = model.connection_pool.schema_reflection
```

Comment:
Can this be nil?

Reply:
No, we always return a `SchemaReflection`. It might be empty though.